halo/evmstaking/keeper/keeper.go

```diff
@@ -283,6 +297,39 @@ func (k Keeper) deliverDelegate(ctx context.Context, ev *bindings.StakingDelegat
 	return nil
 }
 
+// deliverUndelegate processes an Unelegate event.
+func (k Keeper) deliverUndelegate(ctx context.Context, ev *bindings.StakingUndelegate) error {
+	if ev.Amount == nil {
+		return errors.New("unstake amount missing")
+	}
+
+	delAddr := sdk.AccAddress(ev.Delegator.Bytes())
+	valAddr := sdk.ValAddress(ev.Validator.Bytes())
+
+	if _, err := k.sKeeper.GetValidator(ctx, valAddr); err != nil {
+		return errors.New("validator does not exist", "validator", valAddr.String())
+	}
+
+	amountCoin, _ := omniToBondCoin(ev.Amount)
+
+	if !k.hasAccount(ctx, delAddr) {
+		return errors.New("no account", "delegator", delAddr.String())
+	}
```

Comment:
nit: do we really need these two checks? I'm pretty sure `sServer.Undelegate` will fail if the validator or delegator doesn't exist. Isn't that sufficient?